specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/cluster.json

```diff
@@ -1933,6 +1956,148 @@
         }
       }
     },
+    "PrivateLinkConfiguration": {
```

Comment:
>PrivateLinkConfiguration [](http://example.com/codeflow?start=5&length=24)

You might want to consider using a $ref to a common definition for Resource to get id, name, and type. There is one in common-types, for example. This could be used in several of your new definitions.

Reply:
> > PrivateLinkConfiguration
> 
> You might want to consider using a $ref to a common definition for Resource to get id, name, and type. There is one in common-types, for example. This could be used in several of your new definitions.

I can't. Actually I tried. The difference is that we need to make `name` of PrivateLinkConfiguration is settable. If I use $ref to reference a Resource definition, the name will be readonly then in c# SDK, customer can not assign value to name. However our service require the name, we need to support customer assign value to name property in sdk